python/openmldb_sdk/openmldb/dbapi/dbapi.py

```diff
@@ -291,44 +291,50 @@ def execute(self, operation, parameters=()):
 
             self._pre_process_result(rs)
             return self
-        
+
         else:
             ok, rs = self.connection._sdk.execute(command)
             if not ok:
                 raise DatabaseError(rs)
-            
+
             self._pre_process_result(rs)
             return self
 
     @classmethod
-    def __get_append_map(cls, builder, row, hole_idxes, schema):
+    def __get_str_size(cls, row, hole_idxes, schema):
         # calc str total length
         str_size = 0
         for i in range(len(hole_idxes)):
             idx = hole_idxes[i]
             name = schema.GetColumnName(idx)
+            col_value = None
+            col_type = schema.GetColumnType(idx)
+            if col_type != sql_router_sdk.kTypeString:
+                continue
             if isinstance(row, tuple):
-                if isinstance(row[i], str):
-                    str_size += len(row[i])
+                col_value = row[i]
             elif isinstance(row, dict):
                 if name not in row:
                     raise DatabaseError("col {} data not given".format(name))
                 if row[name] is None:
                     if schema.IsColumnNotNull(idx):
```

Comment:
tuple need check not null too?

Reply:
ok，updated

Reply:
> ok，updated

I don't see any change about check not null when row is an instance of tuple, you only check when row is dict

Reply:
updated, please review

Reply:
> > ok，updated
> 
> I don't see any change about check not null when row is an instance of tuple, you only check when row is dict

updated, please review